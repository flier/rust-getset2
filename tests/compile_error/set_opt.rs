use getset2::Setter;

#[derive(Setter)]
pub struct Struct {
    #[set(opt)] // #[set(op)] is not allowed for a field that is not a Option<T>
    field: usize,
}

#[derive(Setter)]
#[set(opt)] // #[set(opt)] is ignored when it applied to the structure
pub struct Struct2 {
    field: usize,
}

fn main() {}
