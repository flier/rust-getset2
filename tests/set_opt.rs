use getset2::{Getter, Setter};

#[derive(Default, Getter, Setter)]
pub struct Foo {
    #[set(opt)]
    option_field: Option<usize>,
}

#[test]
fn set_opt() {
    let mut foo = Foo::default();

    assert_eq!(foo.set_option_field(123).option_field().unwrap(), 123);
}
