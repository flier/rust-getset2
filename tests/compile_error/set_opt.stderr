error: #[set(opt)] should be applied to an `Option<T>` type
 --> tests/compile_error/set_opt.rs:6:12
  |
6 |     field: usize,
  |            ^^^^^
